Bottom: cd52f5e3dbbaf9ea5d91bdfd99cf8c0fef232b65
Top:    1e1be3e31548a3040f15f15b8585c10593699c1f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-22 00:25:27 +0100

Refresh of hid-gtk-allow-zooming-out-past

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index ece47d0..02129e6 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -111,6 +111,7 @@ ghid_zoom_view_rel (Coord center_x, Coord center_y, double factor)
 static void
 ghid_zoom_view_fit (void)
 {
+  ghid_pan_view_abs (0, 0, 0, 0);
   ghid_zoom_view_abs (0, 0, MAX (PCB->MaxWidth  / gport->width,
                                  PCB->MaxHeight / gport->height));
 }
@@ -1567,9 +1568,6 @@ Center(int argc, char **argv, Coord pcb_x, Coord pcb_y)
   widget_x = gport->width / 2;
   widget_y = gport->height / 2;
 
-  printf ("Pointer_x: %i, Pointer_y: %i\n", widget_x, widget_y);
-  pcb_printf ("PCB coords: %#mD\n", pcb_x, pcb_y);
-
   ghid_pan_view_abs (pcb_x, pcb_y, widget_x, widget_y);
 
   /* Now move the mouse pointer to the place where the board location
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 25b4ec4..eff28bc 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -77,28 +77,8 @@ ghid_port_ranges_pan (gdouble x, gdouble y, gboolean relative)
   x0 = h_adj->value;
   y0 = v_adj->value;
 
-  if (relative)
-    {
-      x1 = x0 + x;
-      y1 = y0 + y;
-    }
-  else
-    {
-      x1 = x;
-      y1 = y;
-    }
-
-#if 0
-  if (x1 < h_adj->lower)
-    x1 = h_adj->lower;
-  if (x1 > h_adj->upper - h_adj->page_size)
-    x1 = h_adj->upper - h_adj->page_size;
-
-  if (y1 < v_adj->lower)
-    y1 = v_adj->lower;
-  if (y1 > v_adj->upper - v_adj->page_size)
-    y1 = v_adj->upper - v_adj->page_size;
-#endif
+  x1 = relative ? x + x0 : x;
+  y1 = relative ? y + y0 : y;
 
   ghidgui->adjustment_changed_holdoff = TRUE;
   gtk_range_set_value (GTK_RANGE (ghidgui->h_range), x1);
